blueprint-reference: add Ignition config section

7flying opened the pull request:
Add info about the new ignition customization.

Please wait until osbuild/osbuild-composer#3161 gets merged to merge this (if approved oc).

7flying commented:
Looking for a review on this, the PR that introduced the changes described here has been merged.

7flying commented:
Updated after some updates made on osbuild/osbuild-composer#3130

runcom commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
```diff
@@ -227,6 +227,39 @@ The service names are systemd service units. You may specify any systemd unit fi
+
+The `customizations.ignition` section allows to provide
+[Ignition](https://coreos.github.io/ignition/) configuration files to be used
+in `edge-simplified-installer` ISO images. Check the `r4e`
```
I think ignition.firstboot is also gonna be for edge-raw-image (I'll confirm by end of day)

7flying replied:
Great :+1:

7flying replied:
Do I have to add the `edge-raw-image`part too? @runcom

runcom replied:
I think so yeah, raw image just supports firstboot indeed

elianerpereira commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
```diff
@@ -227,6 +227,39 @@ The service names are systemd service units. You may specify any systemd unit fi
+
+The blueprint configuration can be done *either* by embedding an ignition
+configuration file into the ISO (this can be done *either* by providing a
+Base64 encoded Ignition file or a URL to where the Ignition file configurarion
```
Apologies for commenting here, I notice that there is a small typo in "configurarion". Thank you.

7flying replied:
Your comments are super welcome!
Nice catch :+1:

runcom commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
````diff
@@ -227,6 +227,39 @@ The service names are systemd service units. You may specify any systemd unit fi
+
+```toml
+[customizations.ignition.embedded]
+url = "http://some-server/configuration.ig"
````
also Irene, we need to ditch this one as we're not supporting it

elianerpereira commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
````diff
@@ -227,6 +227,39 @@ The service names are systemd service units. You may specify any systemd unit fi
+url = "http://some-server/configuration.ig"
+```
+
+The way of specifiying an URL with an ignition file to be fetched at first boot
````
I also noticed that there is a small typo in "specifiying". Thank you.

7flying commented:
@runcom @elianerpereira changes made, let me know if it makes sense

miabbott commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
```diff
@@ -227,6 +227,41 @@ The service names are systemd service units. You may specify any systemd unit fi
+
+### Ignition
+
+The `customizations.ignition` section allows to provide
```
I think this reads a little better as :

> The `customizations.ignition` section allows users to provide

7flying commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
```diff
@@ -227,6 +227,41 @@ The service names are systemd service units. You may specify any systemd unit fi
+[butane](https://coreos.github.io/butane/specs/) specification for a
+description of the supported configuration options.
+
+The blueprint configuration can be done *either* by embedding an ignition
```
Yep, if fact everywhere else is on capital letters, so that's a slip on my end, nice catch :+1:

7flying commented:
Updated after @miabbott comments.

thozza commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
```diff
@@ -293,6 +293,40 @@ data = "Hello world!"
+
+The `customizations.ignition` section allows users to provide
+[Ignition](https://coreos.github.io/ignition/) configuration files to be used
+in `edge-raw-image` and `edge-simplified-installer` ISO images. Check the `r4e` 
```
Would it make sense to specify "RHEL for Edge (r4e)" instead of just "r4e"?

thozza replied:
In addition to that, I would omit the "ISO" from this line. The reason is that the `edge-raw-image` is not an ISO and the sentence make is look as if it was (IMO).

thozza commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
````diff
@@ -293,6 +293,40 @@ data = "Hello world!"
+
+```toml
+[customizations.ignition.embedded]
+config = "eyJ --- BASE64 STRING TRIMMED --- 19fQo="
````
I would just specify a valid `base64` string. Ideally some very short and simple configuration.

thozza commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
```diff
@@ -293,6 +293,40 @@ data = "Hello world!"
+description of the supported configuration options.
+
+The blueprint configuration can be done *either* by embedding an Ignition
+configuration file into the ISO (only available for
```
I suggest to replace "ISO" with "image".

thozza reviewed (changes requested):
Thanks for the PR. I added a few comments. In addition to that, please do not break lines in the same paragraph and let the MD interpreter which generates the Guides Web page deal with that. IOW, a paragraph should be a single long line.

7flying commented:
Replaced ISO with image in a couple of places, added full base64 working example, changed paragraphs to be single-line paragraphs, expanded the `r4e` acronym

thozza commented on osbuild-composer/src/blueprint-reference/blueprint-reference.md:
````diff
@@ -293,6 +293,30 @@ data = "Hello world!"
+config = "eyJpZ25pdGlvbiI6eyJ2ZXJzaW9uIjoiMy4zLjAifSwicGFzc3dkIjp7InVzZXJzIjpbeyJncm91cHMiOlsid2hlZWwiXSwibmFtZSI6ImNvcmUiLCJwYXNzd29yZEhhc2giOiIkNiRqZnVObk85dDFCdjdOLjdrJEhxUnhxMmJsdFIzek15QUhqc1N6YmU3dUJIWEVyTzFZdnFwaTZsamNJMDZkUUJrZldYWFpDdUUubUpja2xQVHdhQTlyL3hwSmlFZFdEcXR4bGU3aDgxIn1dfX0="
+```
+
+Add a `base64` encoded Ignition configuration in the `config` field. This Ignition configuration will be included in the `edge-simplified-installer` ISO.
````
Sorry, misses one instance of "ISO" here. I would suggest to also replace it with "image" if that makes sense. My reasoning is that the Ignition customization is applied to the payload installed by the installer, not the installer ISO itself (if I'm not mistaken).

thozza reviewed (commented):
Sorry, but I missed one thing in my previous review. Otherwise this looks OK from my PoV

thozza reviewed (dismissed):
👍

7flying commented:
@supakeen thank you!!
